api

```diff
@@ -283,9 +283,9 @@ remove_repofile_if_unused() { #Given a sources.list.d file, delete it if nothing
       # determine what uri, suite, and components are in a file
       #each stanza can only have one matching section. if there are multiple matches the last one is used
       #case should be ignored for fields
-      local uri="$(sed -n "$line_start","$line_end"p  "$file" | grep -v '^#' | grep -i '^URIs: ' | sed 's/URIs: //Ig' | tail -1)"
-      local suites="$(sed -n "$line_start","$line_end"p "$file" | grep -v '^#' | grep -i '^Suites: ' | sed 's/Suites: //Ig' | tail -1)"
-      local components="$(sed -n "$line_start","$line_end"p "$file" | grep -v '^#' | grep -i '^Components: ' | sed 's/Components: //Ig' | tail -1)"
+      local uri="$(sed -n "$line_start","$line_end"p  "$file" | grep -v '^#' | grep -i '^URIs:' | sed 's/URIs://Ig' | awk '{print $1}' | tail -1)"
+      local suites="$(sed -n "$line_start","$line_end"p "$file" | grep -v '^#' | grep -i '^Suites:' | sed 's/Suites://Ig' | awk '{$1=$1};1' | tail -1)"
+      local components="$(sed -n "$line_start","$line_end"p "$file" | grep -v '^#' | grep -i '^Components:' | sed 's/Components://Ig' | awk '{$1=$1};1' | tail -1)"
```

Comment:
Why awk print 1 on the first line but something else for the next 2?

Reply:
see the commit message from https://github.com/Botspot/pi-apps/pull/2554/commits/3f3b3692b6ef4c415c7fc5f184a0083d620dc4cb

Reply:
> see the commit message from [3f3b369](https://github.com/Botspot/pi-apps/commit/3f3b3692b6ef4c415c7fc5f184a0083d620dc4cb)

I already read that. What I meant was, why does line 286 use `awk '{print $1}'` but lines 287 and 288 use `awk '{$1=$1};1'`?

Reply:
It would seem to me that `awk '{$1=$1};1'` could be used for line 286, and if not, you likely have a reason why which I do not see stated clearly anywhere.

Reply:
> I already read that. What I meant was, why does line 286 use `awk '{print $1}'` but lines 287 and 288 use `awk '{$1=$1};1'`?

see again:
> components and suites that can have multiple strings of output

meanwhile uri can only have one string of output. this is defined by the specification referenced in a comment https://github.com/Botspot/pi-apps/pull/2554/files#diff-14c2529eb4498c5d1ffd6915d05bf58a91bdda796af59f41d480d11c099d0479R264 and expanded upon in an explanation of the specification https://repolib.readthedocs.io/en/latest/deb822-format.html#uris

Reply:
ah hmm... looks like I missed something in the spec. it appears actually that `URIs:` can have multiple uris defined in one field. I guess that explains why it is plural. it will need a loop as well.

> The URI must specify the base of the Debian distribution archive, from which APT finds the information it needs. There must be a URI component present in order for the source to be valid; multipls URIs can be configured simultaneously by adding a space-separated list of URIs.

I will fix that later.